src/Renderer/Shader/TileFS.glsl

```diff
@@ -122,21 +122,22 @@ void main() {
                             textureIndex,
                             projWGS84 ? vUv_WGS84 : uvPM);
 
-                        if (layerColor.a > 0.0) {
+                        bool isVisible = layerColor.a > 0.0;
+                        // if first texture is transparent, we need to force blending with first background color (CWhite)
+                        if ((isVisible && paramsA.w > 0.0) || !validTexture) {
                             validTexture = true;
-                            float lum = 1.0;
-
-                            if(paramsA.z > 0.0) {
-                                float a = max(0.05,1.0 - length(layerColor.xyz-CWhite.xyz));
-                                if(paramsA.z > 2.0) {
-                                    a = (layerColor.r + layerColor.g + layerColor.b)*0.333333333;
-                                    layerColor*= layerColor*layerColor;
-                                }
-                                lum = 1.0-pow(abs(a),paramsA.z);
+                            if(paramsA.z > 2.0) {
+                                float a = max(0.05,1.0 - length(layerColor.xyz - CWhite.xyz));
+                                layerColor.rgb *= layerColor.rgb * layerColor.rgb;
+                                layerColor.a *= 1.0 - pow(abs(a), paramsA.z);
+                            } else if(paramsA.z > 0.0) {
+                                float a = (layerColor.r + layerColor.g + layerColor.b) * 0.333333333;
+                                layerColor.a *= 1.0 - pow(abs(a), paramsA.z);
                             }
 
-                            diffuseColor = mix( diffuseColor,layerColor, lum*paramsA.w * layerColor.a);
-
+                            // In formula, Premultiply-alpha is used to avoid gl_linear filtering artifact.
```

Comment:
What about using this comment instead:
```js
// Use premultiplied-alpha blending formula because source textures are either:
//     - fully opaque (layer.transparent = false)
//     - or use premultiplied alpha (texture.premultiplyAlpha = true)
// Note: using material.premultipliedAlpha doesn't make sense since we're manually blending
// the multiple colors in the shader.
```

Reply:
ok